client/dvm/consensus/src/lib.rs

```diff
@@ -128,24 +132,44 @@ where
 			let log = find_frontier_log::<B>(&block.header)?;
 			let hash = block.post_hash();
 
-			match log {
-				ConsensusLog::EndBlock {
-					block_hash,
-					transaction_hashes,
-				} => {
-					aux_schema::write_block_hash(
-						client.as_ref(),
-						block_hash,
-						hash,
-						insert_closure!(),
-					);
-
-					for (index, transaction_hash) in transaction_hashes.into_iter().enumerate() {
-						aux_schema::write_transaction_metadata(
-							transaction_hash,
-							(block_hash, index as u32),
-							insert_closure!(),
-						);
+			let post_hashes = match log {
+				ConsensusLog::PostHashes(post_hashes) => post_hashes,
+				ConsensusLog::PreBlock(block) => {
+					dvm_consensus_primitives::PostHashes::from_block(block)
+				}
+				ConsensusLog::PostBlock(block) => {
+					dvm_consensus_primitives::PostHashes::from_block(block)
+				}
+			};
+
+			let mapping_commitment = dvm_db::MappingCommitment {
+				block_hash: hash,
+				ethereum_block_hash: post_hashes.block_hash,
+				ethereum_transaction_hashes: post_hashes.transaction_hashes,
+			};
+			let res = self.backend.mapping_db().write_hashes(mapping_commitment);
+			if res.is_err() {
+				trace!(target: "frontier-consensus", "{:?}", res);
+			}
```

Comment:
There are changes related to client db here. Do we need to do client side db migration when we incorporate dvm into Crab Network?

Reply:
At this stage, the crab network does not have an integrated DVM solution. I don't think migration is needed if the upgraded database solution is chosen directly when integrating DVM to crab, right? But there is another problem, if the DVM already running on crab has data incompatibility in the future, the way of data migration may be different from the way of substrate migration, which needs further thought.

Reply:
Let's do such a test on Pangolin.

1. Upgrade to simulate supporting new added DVM to Pangolin. (Clean DVM from Pangolin before that)
2. DVM already running on pangolin, do client backend migration if there are incompatibility datas or upgrades from Frontier.

Reply:
#512 here.